bindings/rust/s2n-tls-sys/build.rs

```diff
@@ -99,12 +99,16 @@ fn build_vendored() {
         .flag_if_supported("-z noexecstack")
         // we use some deprecated libcrypto features so don't warn here
         .flag_if_supported("-Wno-deprecated-declarations")
-        .define("_POSIX_C_SOURCE", "200112L");
+        .define("_POSIX_C_SOURCE", "200112L")
+        .define("_FORTIFY_SOURCE", "2");
 
-    if env("PROFILE") == "release" {
-        // fortify source is only available in release mode
-        build.define("_FORTIFY_SOURCE", "2");
+    // Enabling _FORTIFY_SOURCE requires at least some optimizations to be enabled so use level 1
+    // at the minimum.
+    if env("OPT_LEVEL") == "0" {
+        build.opt_level(1);
+    }
```

Comment:
Some versions of openssl seem to be shipping with `_FORTIFY_SOURCE` defined in the headers so I just enabled optimizations to bypass the warnings. Otherwise you get a massive list of locations:

```
warning: /nix/store/l1a7p8yp0nlvghq5vifz4qr6s11frxl5-glibc-2.35-224-dev/include/features.h:412:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
warning:   412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
warning:       |    ^~~~~~~
warning: In file included from /nix/store/l1a7p8yp0nlvghq5vifz4qr6s11frxl5-glibc-2.35-224-dev/include/bits/libc-header-start.h:33,
warning:                  from /nix/store/l1a7p8yp0nlvghq5vifz4qr6s11frxl5-glibc-2.35-224-dev/include/stdint.h:26,
warning:                  from /nix/store/gwm7zfz09wibym9hiqckdjj1dndzbmi0-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include/stdint.h:9,
warning:                  from lib/api/s2n.h:42,
warning:                  from lib/crypto/s2n_evp_signing.h:18,
warning:                  from lib/crypto/s2n_evp_signing.c:16:
warning: /nix/store/l1a7p8yp0nlvghq5vifz4qr6s11frxl5-glibc-2.35-224-dev/include/features.h:412:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
warning:   412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
warning:       |    ^~~~~~~
warning: In file included from /nix/store/l1a7p8yp0nlvghq5vifz4qr6s11frxl5-glibc-2.35-224-dev/include/bits/libc-header-start.h:33,
warning:                  from /nix/store/l1a7p8yp0nlvghq5vifz4qr6s11frxl5-glibc-2.35-224-dev/include/limits.h:26,
warning:                  from /nix/store/gwm7zfz09wibym9hiqckdjj1dndzbmi0-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/limits.h:203,
warning:                  from /nix/store/gwm7zfz09wibym9hiqckdjj1dndzbmi0-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/syslimits.h:7,
warning:                  from /nix/store/gwm7zfz09wibym9hiqckdjj1dndzbmi0-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/limits.h:34,
warning:                  from /home/cameron/Projects/aws/s2n-tls/bindings/rust/target/debug/build/openssl-sys-3597dc570ecf585b/out/openssl-build/install/include/openssl/ossl_typ.h:13,
warning:                  from /home/cameron/Projects/aws/s2n-tls/bindings/rust/target/debug/build/openssl-sys-3597dc570ecf585b/out/openssl-build/install/include/openssl/evp.h:14,
warning:                  from lib/crypto/s2n_cipher.c:16:
...
```

Reply:
Is this related to Nix? Should we open an issue to track this problem? Could make debugging the bindings harder

Reply:
I can do some more investigation. It's possible that it's just a Nix issue...

Reply:
Looks like it is a Nix issue: https://github.com/NixOS/nixpkgs/issues/18995

I will revert this change for now to unblock the release.